### PR TITLE
Add on-device test retries for android and raspi

### PR DESCRIPTION
**Pull request by oxve**
b/279935125

### CODE REVIEW

File: .github/actions/on_device_tests/action.yaml [hunk 1]
```diff
@@ -42,6 +42,14 @@ runs:
           echo "TEST_TYPE=unit_test" >> $GITHUB_ENV
           echo "USE_SHARDING=1" >> $GITHUB_ENV
         fi
+
+        if [[ "${{ matrix.target_platform }}" == "android"* ]]; then
+          # Test retries for android
+          echo "TEST_ATTEMPTS=3" >> $GITHUB_ENV
```
Comment (isarkis) [inline .github/actions/on_device_tests/action.yaml:48]: I think we should put TEST_ATTEMPTS configuration into https://github.com/youtube/cobalt/blob/main/.github/config/android-x86.json#L3

File: .github/config/android-arm.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
       "2",
       "3",
       "black_box_test"
-    ]
+    ],
+    "test_attempts": 3
```
Comment (kaidokert) [inline .github/config/android-arm.json:12]: I'd actually be a bit conservative and go with 2 out of the gate as a default. Device related errors haven't been too frequent recently, and I'm a bit cautious about just adding extra feedback delay if something is _really broken_.

If the normal test run gives you "its broken" signal in 15 minutes, now you have to wait 45 minutes.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -267,7 +270,8 @@ jobs:
         shard: ${{ fromJson(needs.initialize.outputs.on_device_test).tests }}
         include: ${{ fromJson(needs.initialize.outputs.includes) }}
     env:
-      COBALT_BOOTLOADER: ${{needs.initialize.outputs.bootloader}}
+      COBALT_BOOTLOADER: ${{ needs.initialize.outputs.bootloader }}
```
Comment (kaidokert) [inline .github/workflows/main.yaml:273]: I thought we had pre-commit yaml formatter enabled by default, no ?

File: tools/on_device_tests_gateway_client.py [hunk 4]
```diff
@@ -222,6 +224,18 @@ def main():
...
+      type=str,
+      default='1',
+      required=False,
+      help='The maximum number of times a test could retry.')
+  trigger_parser.add_argument(
+      '--retry_level',
+      type=str,
+      default='ERROR',
+      required=False,
+      help='The retry level of Mobile harness job. Either ERROR or FAIL.')
```
Comment (kaidokert) [inline tools/on_device_tests_gateway_client.py:238]: Could we leave a bit more context here as to what ERROR or FAIL mean ? I mean i can look them up on the doc page separately, but to a casual reader they sort of look similar. tl;dr i want to ensure we only retry when we get infra errors, not when we get actual distinguishable test failures

**Review by kaidokert — APPROVED** [review]

File: .github/actions/on_device_tests/action.yaml [hunk 5]
```diff
@@ -74,6 +74,7 @@ runs:
             ${LOADER_PLATFORM:+"--loader_platform" "$LOADER_PLATFORM"} \
             ${DIMENSION:+"--dimension" "$DIMENSION"} \
             ${USE_SHARDING:+"--unittest_shard_index" "${{ matrix.shard }}"} \
+            ${TEST_ATTEMPTS:-"--test_attempts" "$TEST_ATTEMPTS"} \
```
Comment (isarkis) [inline .github/actions/on_device_tests/action.yaml:77]: I think this should be ${TEST_ATTEMPTS:+"--test_attempts" "$TEST_ATTEMPTS"} vs ${TEST_ATTEMPTS:-"--test_attempts" "$TEST_ATTEMPTS"} ?

File: .github/workflows/main.yaml (hunk 3 above)
Comment (oxve) [reply .github/workflows/main.yaml:273]: It seems it does not format these expressions. @isarkis maybe you know?

File: .github/actions/on_device_tests/action.yaml (hunk 5 above)
Comment (oxve) [reply .github/actions/on_device_tests/action.yaml:77]: Yes, I think you are right. I tested this a bit more locally and found that it should work even if the variable isset to empty string, which I think is the case if the field is not in the platform config.

I also renamed the variable to be clear that it's specific to on device tests.

File: .github/workflows/main.yaml (hunk 3 above)
Comment (isarkis) [reply .github/workflows/main.yaml:273]: I don't think our yaml linter can handle GitHub Actions specific yaml syntax.

**Review by isarkis — APPROVED** [review]

**Comment by oxve** [comment]
Parameters are passed to MH as expected:
```
DIMENSIONS:
- platform=android-x86
JOB TIMEOUT:	41400 sec
TEST_TIMEOUT:	5100 sec
START_TIMEOUT:	36000 sec
TEST_ATTEMPTS:	2
RETRY_LEVEL:	ERROR
PRIORITY:	DEFAULT
```

Test failures are unrelated.

**Pull request merged**